validator/node/node.go

```diff
@@ -249,22 +251,50 @@ func (s *ValidatorClient) initializeFromCLI(cliCtx *cli.Context) error {
 	return nil
 }
 
-func moveDb(cliCtx *cli.Context, accountsDir string) string {
+func (s *ValidatorClient) moveDb(cliCtx *cli.Context, dbDir string) string {
 	dataDir := cliCtx.String(cmd.DataDirFlag.Name)
-	if accountsDir != "" {
+	if dbDir != "" {
 		dataFile := filepath.Join(dataDir, kv.ProtectionDbFileName)
-		newDataFile := filepath.Join(accountsDir, kv.ProtectionDbFileName)
-		if fileutil.FileExists(dataFile) && !fileutil.FileExists(newDataFile) {
+		var accountsDirDbFile string
+		if s.wallet != nil {
+			accountsDirDbFile = filepath.Join(s.wallet.AccountsDir(), kv.ProtectionDbFileName)
+		}
+		newDataFile := filepath.Join(dbDir, kv.ProtectionDbFileName)
+		accountsDBExistAndDifferent := accountsDirDbFile != "" && accountsDirDbFile != newDataFile
+		accountsDBExistsNoDataDirDb := fileutil.FileExists(accountsDirDbFile) && !fileutil.FileExists(newDataFile)
+		dataDirDBExistsAndDifferent := newDataFile != dataFile && fileutil.FileExists(dataFile)
+		accountsDirDBDoesntExist := !fileutil.FileExists(newDataFile)
+		if accountsDBExistAndDifferent && accountsDBExistsNoDataDirDb { //move files from account dir to datadir.
 			log.WithFields(logrus.Fields{
-				"oldDbPath": dataDir,
-				"walletDir": accountsDir,
-			}).Info("Moving validator protection db to wallet dir")
-			err := fileutil.CopyFile(dataFile, newDataFile)
+				"oldDbPath":      s.wallet.AccountsDir(),
+				"validatorDbDir": dbDir,
+			}).Info("Moving validator protection db from wallet dir")
+			err := fileutil.CopyFile(accountsDirDbFile, newDataFile)
 			if err != nil {
 				log.Fatal(err)
 			}
+			log.WithFields(logrus.Fields{
+				"oldDbFile": dataFile,
+			}).Info("Deleting validator db in folder")
+			if err := os.Remove(accountsDirDbFile); err != nil {
+				log.Info(errors.Wrap(err, "could not delete old db file"))
+			}
+		} else if dataDirDBExistsAndDifferent && accountsDirDBDoesntExist { //move files to account dir.
```

Comment:
hmm no, it means the user wants to use a db from that directory. Copying and removing it isnt correct